packages/trace-viewer/src/ui/llm.tsx

```diff
@@ -249,12 +249,13 @@ export function useIsLLMAvailable() {
   return !!React.useContext(llmContext);
 }
 
-export function useLLMConversation(id: string) {
-  const conversation = useLLMChat().getConversation(id);
-  if (!conversation)
-    throw new Error('No conversation found for id: ' + id);
-  const [history, setHistory] = React.useState(conversation.history);
+export function useLLMConversation(id?: string): readonly [LLMMessage[], Conversation | undefined] {
+  const chat = useLLMChat();
+  const conversation = id ? chat.getConversation(id) : undefined;
+  const [history, setHistory] = React.useState(conversation?.history ?? []);
   React.useEffect(() => {
+    if (!conversation)
+      return;
     function update() {
       setHistory([...conversation!.history]);
```

Comment:
Unlikely to occur, but you should remove the force unwrap here and add a second `!conversation` check.

Reply:
The _reference_ is part of the closure. It is not a value type. The `history` property can change at any time, ans since `update` is run async, it needs to be checked.

Reply:
You're right. For some reason when I came back to this I thought the force unwrap was on `history` for some reason.